### DISCUSSION TITLE
DynamoDB avoid to insert values when they are None

### DISCUSSION THREAD

**Discussion by ymwjbxxq** [awslabs/aws-sdk-rust#538] [Q&A] [answered]
Hello,

imagine this struct:

```
struct Person {
    name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    surname: Option<String>,
}
```

If I POST my JSON:
```
{
  name: daniele
}
```

I would like to see it in DynamoDB as:
```
{
 "name": {
  "S": "daniele"
 }
}
```

At the moment, I have to default always the value because:

```
 .expression_attribute_values(":name",AttributeValue::S(request.name.clone()))
.expression_attribute_values(":surname",AttributeValue::S(request.surname.clone().unwrap_or("".to_string())))
```

And I end up with:
```
{
 "name": {
  "S": "daniele"
 },
"surname": {
  "S": ""
 }
}
```

Is there any way to do that? Like an option to configure to declare to skip the None?

Of course, I can do this:
```
 let mut update_expression: String = "SET name = :name".to_owned();
 if request.surname.is_some() {
        query = query
                    .expression_attribute_values(":surname",AttributeValue::S(request.surname.clone().unwrap()));
       update_expression.push_str(", surname = :surname ");
      }
```

But imagine a big struct. I just wish or would like to know if there is a way to instruct the SDK to skip None and have much less code around.

Thanks,
Dan

**Comment by jdisanti** [comment, marked as answer]
I don't think there is a great non-verbose solution for this right now since we haven't implemented any kind of high level DynamoDB library (see discussion on https://github.com/awslabs/aws-sdk-rust/issues/70). In the meantime, I know of at least [one library in crates.io](https://crates.io/crates/serde_dynamo) that does this for you. Otherwise, you may be able to move the bulk of this conditional logic into a macro.

**Reply by ymwjbxxq** [reply]
Thanks for the feedback. I would like something aws official and not another library even if it comes from serde people.
I wish to have the SDK more helpful and follow the official way.


Could you point me to some macro examples that I can use? I am not very familiar with that.
